### PR TITLE
Feature/image formats/console

### PR DESCRIPTION
**Pull request by YarikTalko**

### CODE REVIEW

File: ImageConverter/ImageConverter/Input.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
+﻿using ImageConverter.Bmp;
+using ImageConverter.Ppm;
+
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:3]: also gif

File: ImageConverter/ImageConverter/Input.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        public void InputCheck(string[] args)
+        {
+            string source = "";
+            string destination = "";
+            string goal_format = "";
+            bool isDestinationParametеrExist = false;
+            bool isSourceParameterExist = false;
+            bool isGoalFormatParameterExist = false;
+
+            if (args.Length > 6)
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:22]: what if too few parameters?

File: ImageConverter/ImageConverter/Input.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                if (args[i] == "--goal-format")
+                {
+                    goal_format = args[i + 1];
+                    isGoalFormatParameterExist = true;
+                }
+            }
+
+            if (!isDestinationParametеrExist)
+            {
+                throw new Exception("you forget --destination");
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:48]: why not ArgumentException or IOException etc?

File: ImageConverter/ImageConverter/Input.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            }
+            if (!isSourceParameterExist)
+            {
+                throw new Exception("you forget --source");
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:52]: forgot*

File: ImageConverter/ImageConverter/Input.cs (hunk 3 above)
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:48]: > `--output` - шлях до результату (за замовчуванням використовується ім'я та шлях оригінального файлу з новим розширенням)

File: ImageConverter/ImageConverter/Input.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            }
+            if (!isGoalFormatParameterExist)
+            {
+                throw new Exception("you forget --goal-format");
+            }
+            if (!File.Exists(source))
+            {
+                throw new Exception("source is not exists");
+            }
+            if (Path.GetExtension(destination) != ".bmp" && Path.GetExtension(destination) != ".ppm")
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:62]: should be `source` instead of `destination` I guess

Comment (Melkye) [reply ImageConverter/ImageConverter/Input.cs:62]: also add gif

File: ImageConverter/ImageConverter/Input.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                throw new Exception("destination is wrong");
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:64]: tell explicitly why it's wrong
i. e. "only ppm, bmp and gif are supported as source'

File: ImageConverter/ImageConverter/Input.cs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            }
+            if (goal_format != ".bmp" && goal_format != ".ppm")
+            {
+                throw new Exception("goal-format must be .ppm or .bmp");
+            }
+            if (Path.GetExtension(destination) == Path.GetExtension(source))
+            {
+                throw new Exception("destination file extension cannot be the same as source file extension");
+            }
+            if (Path.GetExtension(destination) != goal_format)
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:74]: should be removed since destination is a folder

File: ImageConverter/ImageConverter/Input.cs [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                throw new Exception("destination file extension must be the same as goal-format");
+            }
+            if (string.IsNullOrEmpty(destination) || string.IsNullOrEmpty(source) || string.IsNullOrEmpty(goal_format))
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:78]: why don't check it at first?

Comment (Melkye) [reply ImageConverter/ImageConverter/Input.cs:78]: they even can't be null cuz u split by spaces

File: ImageConverter/ImageConverter/Input.cs [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                throw new Exception("destination, source and goal-format must not be null or empty");
+            }
+
+            Console.WriteLine(destination + " " + source + " " + goal_format);
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:83]: what is it for? and why it's exactly here?

File: ImageConverter/ImageConverter/Input.cs [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+namespace ImageConverter
+{
+    internal class Input
+    {
+        public Input()
+        {
+
+        }
+
+        public void InputCheck(string[] args)
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:13]: you name it "check" but then you call actual converter, need to separate those

File: ImageConverter/ImageConverter/Input.cs (hunk 2 above)
Comment (Melkye) [reply ImageConverter/ImageConverter/Input.cs:22]: what if only the key is specified but not the value?

File: ImageConverter/ImageConverter/Input.cs [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+            Choise(destination, source, goal_format);
+
+        }
+
+        private void Choise(string destination, string source, string goal_format)
+        {
+            ImageConverter ic;
+
+            if (goal_format == ".bmp")
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:93]: need to add gif

File: ImageConverter/ImageConverter/Input.cs [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+                throw new Exception("destination, source and goal-format must not be null or empty");
+            }
+
+            Console.WriteLine(destination + " " + source + " " + goal_format);
+
+            Choise(destination, source, goal_format);
+
+        }
+
+        private void Choise(string destination, string source, string goal_format)
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:89]: silly name

File: ImageConverter/ImageConverter/Program.cs [hunk 13]
```diff
@@ -1,54 +1,12 @@
 ﻿using ImageConverter;
-using ImageConverter.Bmp;
-using ImageConverter.Ppm;
 
 internal class Program
 {
-    private static void Main(string[] args) // --source smth --goal-format smth --output smth
+    private static void Main(string[] args)
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Program.cs:5]: add exception handing

File: ImageConverter/ImageConverter/Program.cs [hunk 14]
```diff
@@ -1,54 +1,12 @@
...
 internal class Program
 {
-    private static void Main(string[] args) // --source smth --goal-format smth --output smth
+    private static void Main(string[] args)
     {
-        string source = "";
-        string destination = "";
-        string goal_format = "";
+        Input input = new Input();
+        input.InputCheck(args);
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Program.cs:8]: it says nothing that it actually converts the pic

File: ImageConverter/ImageConverter/Program.cs [hunk 15]
```diff
@@ -1,54 +1,12 @@
...
-        //source = "C:\\Users\\arikt\\Documents\\Programming\\ComputerGraphics\\ImageConverter\\Images\\MARBLES.ppm";
-        //destination = "C:\\Users\\arikt\\Documents\\Programming\\ComputerGraphics\\ImageConverter\\Images\\output1.bmp";
-
-        // TODO: create converter based on passed args
-        // NOTE: consider using converter factory
-
-        ImageConverter.ImageConverter ic = new(new PpmImageReader(), new BmpImageWriter());
-
-        ic.Convert(source, goal_format, destination);
+        string bastePath = System.IO.Directory.GetCurrentDirectory();
```
Comment (Melkye) [inline ImageConverter/ImageConverter/Program.cs:10]: ???

File: ImageConverter/ImageConverter/Input.cs (hunk 3 above)
Comment (Melkye) [inline ImageConverter/ImageConverter/Input.cs:48]: It'd be better to throw an exception based on not only one problem but on all of them with a message containing all encountered input mistakes

**Review by Melkye — CHANGES_REQUESTED** [review]

File: ImageConverter/ImageConverter/Input.cs (hunk 9 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Input.cs]: sorry, it is for debugging

File: ImageConverter/ImageConverter/Input.cs (hunk 6 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Input.cs]: but I don't have gif

File: ImageConverter/ImageConverter/Input.cs (hunk 1 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Input.cs:3]: I don't have gif

File: ImageConverter/ImageConverter/Input.cs (hunk 7 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Input.cs:78]: no distanse is "C:\\Users\\arikt\\Documents\\Programming\\ComputerGraphics\\ImageConverter\\Images\\output23.ppm" where output23 is the name of file and ppm is the file extension

File: ImageConverter/ImageConverter/Input.cs (hunk 11 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Input.cs:94]: i don`t have gif

File: ImageConverter/ImageConverter/Program.cs (hunk 13 above)
Comment (YarikTalko) [reply ImageConverter/ImageConverter/Program.cs:5]: check program now, not actual

File: ImageConverter/ImageConverter/Input.cs (hunk 1 above)
Comment (Melkye) [reply ImageConverter/ImageConverter/Input.cs:4]: you merged gif branch into this one, wtf

**Pull request merged**